components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -2238,7 +2238,15 @@ public static NativeArray jsFunction_getAPI(Context cx, Scriptable thisObj,
                 PrivilegedCarbonContext.startTenantFlow();
                 PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
             }
-            API api = apiProvider.getAPI(apiId);
+
+            API api = null;
+            try {
+                api = apiProvider.getAPI(apiId);
+            } catch (APIManagementException e) {
+                handleException("Cannot find the requested API- " + apiName +
```

Comment:
Original exception trace is already fixed in commit: https://github.com/wso2/carbon-apimgt/commit/e433e625a5bed755ccab07851ac9510361b9e400